csrc/ir/base_nodes.h

```diff
@@ -241,7 +244,7 @@ class TORCH_CUDA_CU_API Val : public Statement {
       : Val(passkey,
             ValType::Others,
             nvfuser::getDataType(value),
-            std::move(value)) {}
```

Comment:
This is a use-after-move bug. Because the order of argument is undefined, it is possible that the `std::move` is evaluated earlier than the `nvfuser::getDataType`.